pkg/cmd/root/root.go

```diff
@@ -9,14 +12,39 @@ import (
 	"github.com/coreeng/corectl/pkg/cmd/tenant"
 	"github.com/coreeng/corectl/pkg/cmd/version"
 	"github.com/coreeng/corectl/pkg/cmdutil/config"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
 	"github.com/spf13/cobra"
 )
 
-func NewRootCmd(cfg *config.Config) *cobra.Command {
+func ConfigureGlobalLogger(logLevelFlag string, logFile *os.File) {
+	logLevel, logLevelParseError := zerolog.ParseLevel(logLevelFlag)
+	if logLevelParseError != nil {
+		logLevel = zerolog.Disabled
+	}
+	zerolog.SetGlobalLevel(logLevel)
+	if logFile == nil && !(logLevel == zerolog.Disabled) {
+		var err error
+		logFile, err = os.OpenFile("corectl.log", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
```

Comment:
I think it simply overwrites local reference instead of writing to the global variable in `run.go` (if it was your intent). Can we the whole logFile handling in this file? There is a possibility to add hooks to commands, you can use it to handle file closing

Reply:
this is no longer needed with phuslu/log